File: python/programmers/lv2_sum_of_contiguous_subarray.py
# 풀이 코드 
ipt = list(map(int, input().split()))

sequence = ipt[:-1]
k = ipt[-1]

sta, end = 0, 0
s = 0
maxL = float("inf")
ans = []

while end < len(sequence):
   s += sequence[end]
   while s >= k and sta <= end:
      if s == k:
         val = end - sta + 1
         if val < maxL:
            maxL = val
            ans = [sta, end]
      s -= sequence[sta]
      sta += 1
   end += 1
print(ans)

# 제출 코드 
def solution(sequence, k):
    answer = []
    sta, end = 0, 0
    s = 0
    maxL = float("inf")

    while end < len(sequence):
       s += sequence[end]
       while s >= k and sta <= end:
          if s == k:
             val = end - sta + 1
             if val < maxL:
                maxL = val
                answer = [sta, end]
          s -= sequence[sta]
          sta += 1
       end += 1
    
    return answer

File: java/Programmers/Lv2_Sum_Of_Contiguous_Subarray.java
package PACKAGE_NAME;

import java.io.IOException;

public class Lv2_Sum_Of_Contiguous_Subarray {

    class Solution {
        int[] answer = new int[2];
        int sta = 0;
        int end = 0;
        int sum = 0;
        int val = 0;
        int minL = Integer.MAX_VALUE;
        public int[] solution(int[] sequence, int k) {

            while (end < sequence.length) {
                sum += sequence[end];

                while (sum >= k && sta <= end) {
                    if (sum == k) {
                        val = end - sta + 1;
                        if (val < minL) {
                            minL = val;
                            answer[0] = sta;
                            answer[1] = end;
                        }
                    }
                    sum -= sequence[sta];
                    sta ++;
                }
                end ++;
            }

            return answer;
        }
    }

    public static void main(String[] args) throws IOException {
        System.out.println("Hello, World!");
    }
}